crates/proof/proof-interop/src/consolidation.rs

```diff
@@ -142,6 +144,29 @@ where
                 .get(chain_id)
                 .ok_or(ConsolidationError::MissingLocalProvider(*chain_id))?;
 
+            let PreState::TransitionState(ref mut transition_state) =
+                self.boot_info.agreed_pre_state
+            else {
+                return Err(ConsolidationError::InvalidPreStateVariant);
+            };
+            let original_optimistic_block = transition_state
+                .pending_progress
+                .iter_mut()
+                .find(|block| block.block_hash == header.hash())
+                .ok_or(MessageGraphError::EmptyDependencySet)?;
+
+            // Filter out all transactions that are not deposits to start.
+            let mut transactions = transactions
+                .into_iter()
+                .filter(|t| !t.is_empty() && t[0] == OpTxType::Deposit)
```

Comment:
What tx type is this? Would be nice to be explicit and use a strong method to check if the tx type is a deposit tx (e.g. `t.is_deposit()`) rather than having to check if empty and the first byte.

Reply:
So the transactions in the transactions trie are already 2718-encoded, meaning to use [`OpTxEnvelope::is_deposit`](https://docs.rs/op-alloy-consensus/latest/op_alloy_consensus/transaction/enum.OpTxEnvelope.html#method.is_deposit), we'd have to decode them just for this check.

I'd prefer not to mainly for the redundancy there, but not super opinionated; They're re-validated later on in the pre-validation execution step. You have a strong pref here?

Reply:
No that's fine. There's a lot of repeated logic here and we've seen the issue in revm (albeit a magic number issue `0x7F` vs `0x7E`). It would be nice to consolidate this filter a bit so it can be unit tested and in one spot rather than duped. Not a blocker.